### PR TITLE
add codeql pipeline

### PR DESCRIPTION
**Pull request by UlisesGascon**
Refs: https://github.com/nodejs/nodejs.org/issues/5659

### CODE REVIEW

**Review by ovflowd — CHANGES_REQUESTED** [review]
We already have CodeQL enabled by default.

**Comment by UlisesGascon** [comment]
> We already have CodeQL enabled by default.

Even if is enabled, it is recommended to include the pipeline.

Captura de pantalla 2023-08-22 a las 12 40 21

_[Current warnings](https://github.com/nodejs/nodejs.org/security/code-scanning/tools/CodeQL/status)_

**Comment by ovflowd** [comment]
Gotcha. That makes sense. Can you add the missing EOF line?

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    strategy:
+      fail-fast: false
+      matrix:
+        language: ["javascript", "typescript"]
+        # CodeQL supports [ $supported-codeql-languages ]
+        # Learn more about CodeQL language support at https://aka.ms/codeql-docs/language-support
+
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@8ca2b8b2ece13480cda6dacd3511b49857a23c09 # v2.5.1
```
Comment (ovflowd) [inline .github/workflows/codeql.yml:44]: I've whitelisted this Action.

**Comment by ovflowd** [comment]
@UlisesGascon can you checkout your branch locally and fix the linting issues?

**Review by ovflowd — APPROVED** [review]

**Comment by ovflowd** [comment]
Seems like CodeQL is failing.

**Comment by UlisesGascon** [comment]
I am researching about [the error](https://github.com/nodejs/nodejs.org/actions/runs/5985234979/job/16237142971):

> Error: Code Scanning could not process the submitted SARIF file:
> CodeQL analyses from advanced configurations cannot be processed when the default setup is enabled

Seems like there is a misconfiguration. Maybe is related to https://github.com/github/codeql-action/issues/1528. 

Current config in Security tab 🤔 

Captura de pantalla 2023-08-26 a las 16 25 41

**Comment by ovflowd** [comment]
@UlisesGascon are you going to fix this PR?

**Review by aymen94 — APPROVED** [review]

**Comment by ovflowd** [comment]
I'm closing this as a stale PR. Please feel free to reopen at your convenience.

**Pull request closed without merging**